src/index.js

```diff
@@ -31,19 +31,44 @@ const closeArgsToError = (code, signal) => {
   return null
 }
 
+const concatBuffer = (buffer) => {
+  if (!(buffer instanceof Array)) {
+    // throw new Error('Expected buffer to be an Array')
+    return
+  } else if (!buffer.length) {
+    return
+  }
+  let result
+  if (typeof buffer[0] === 'string') {
+    result = buffer.join('')
+  } else if (buffer[0] instanceof Buffer) {
+    result = Buffer.concat(buffer)
+  } else {
+    // throw new Error('Unexpected buffer type')
+    return
+  }
+  return result
+}
+
 export default (cmd, args, options = {}) => new Promise((resolve, reject) => {
   const proc = crossSpawn(cmd, args, options)
   let stdout = null
   let stderr = null
   const [ignoreStdout, ignoreStderr] = parseStdioOption(options.stdio)
   if (!ignoreStdout) {
     stdout = []
+    if (options.encoding) {
+      proc.stdout.setEncoding(options.encoding)
```

Comment:
Maybe the encoding defaults to UTF-8 for console streams? But then the buffers wouldn't have worked, I guess.